### DISCUSSION TITLE
Add a 'changed' event

### DISCUSSION THREAD

**Discussion by flowt-au** [graphology/graphology#463] [Ideas]
I am thinking it would be useful to add a 'changed' listener.

My use-case is that I persist the graph after anything changes (using a debounce function). Instead of having an AddEventListeners() function that adds a listener for every possible change, it would be great to have the option of a single 'changed' event that fires whenever any of the others would fire. 

Maybe it could also pass the name of the listener that would otherwise have been fired so one can maybe ignore specific ones (eg 'clear') or do some other action based on that event?

Just a suggestion for a great library.

Thanks,
Murray

**Comment by Yomguithereal** [comment]
Hello @flowt-au, I see your point and this is a good idea but I already discussed this specific point and I won't add a `changed` event to `graphology` for 2 main reasons:

1. This has a performance cost I don't want to pay because:
2. This can be implemented from userland without any hassle by creating a custom functions or even by monkeypatching the graph's prototype (I recommend the custom functions personally, less risky/bad practice)

**Comment by flowt-au** [comment]
@Yomguithereal Thanks Guillaume, yes, I see your point. All good. :-)
